savant/deepstream/pipeline.py

```diff
@@ -298,8 +289,8 @@ def _on_source_caps(
                 time.sleep(5)
 
         with self._source_adding_lock:
-            self._source_id_by_index[source_info.pad_idx] = source_info.source_id
-            self._sources[source_info.source_id] = source_info
+            self._sources.set_src_id(source_info.pad_idx, source_info.source_id)
+            self._sources.set_src_info(source_info.source_id, source_info)
```

Comment:
```
self._sources.set_src_id(source_info.pad_idx, source_info.source_id)
self._sources.set_src_info(source_info.source_id, source_info)
```
-> `self._sources.update_source(source_info)` or `self._sources.update_source(source_info.source_id, source_info)` 
there is no need to split this operation